File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i=0; i<commands.size();i++){
        int start = commands[i][0];
        int end = commands[i][1];
        int index = commands[i][2];
        
        vector<int> newArr; //4만큼
        // start = 2, end = 5
        // 1,2,3,4 -> 5,2,6,3
        for(int j=start-1; j<end;j++){ // 1~5까지 1,2,3,4,
            newArr.push_back(array[j]);
        }    
        
        sort(newArr.begin(), newArr.end());
        answer.push_back(newArr[index-1]);
    }
    
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        // 0부터 1(1-1)까지 자르고 정렬한다음에 2이 위치에 있는거 픽
        for(int i=0; i<commands.length; i++){
            ArrayList<Integer> arr = new ArrayList<>();
            for(int j=commands[i][0]-1; j<commands[i][1]; j++){
                arr.add(array[j]);
            }
            
            Collections.sort(arr);
            answer[i] = arr.get(commands[i][2]-1);
        }
        
        return answer;
    }
}